Tools/Anonymization/main.py

```diff
@@ -0,0 +1,140 @@
+            text = re.sub(re.compile("#.*?\n" ), "", text)
+
+        # remove block line comments such as (/*COMMENT */, ''' COMMENT ''') if user wants to
+        if self.rm_block_comments:
+            self.logger.info("[" + str(file) + "] Removing all occurances of block line comments")
+            text = re.sub(re.compile("/\*.*?\*/",re.DOTALL ) ,"" ,text) 
+            text = re.sub(re.compile("'''.*?\n" ), "", text)
+
+        # anonymises specfic anonymiser types that user wants
+        for key in self.anonDicts:
+            self.logger.info("[" + str(file) + "] Starting to anonymize " + key + " anoymizer type")
+            text = self.regex_anonymizer(text, self.anonDicts.get(key), key, file)
+        return text
+
+    def getHash(self, word):
+        hash_object = hashlib.sha256(word.encode('utf-8'))
+        hex_dig = " " + hash_object.hexdigest()[:16]
+        return hex_dig
+
+    def replace_all(self, text: str, matchs: Iterable[str], file: str) -> str:
+        '''
+        Replace all occurance in matchs in text with a hash
+        '''
+        for match in matchs:
+            hash = self.getHash(match)
+            self.logger.info("[" + str(file) + "] Replacing `" + match + "` with hash `" + hash + "`")
+            text = text.replace(match, hash)
+        self.logger.info("[" + str(file) + "] Completed replacing obtained matches")
+        return text
+
+    def regex_anonymizer(self, text: str, regex: Pattern, type: str, file: str) -> str:
+        '''
+        Anonymize all substring matching a specific regex with a hash
+        '''
+        regex = re.compile(regex, re.IGNORECASE)
+        self.logger.info("[" + str(file) + "] Finding all occurances of " + type + " anoymizer type")
+        matchs = re.findall(regex, text)
+        return self.replace_all(text, matchs, file)
+
+def main():
+
+    # parser options
+    parser = argparse.ArgumentParser(description="Anonymizer")
+    parser.add_argument("--add", nargs='+', default="all", help="Add anonymizer types")
+    parser.add_argument("--rm_single_comments", default=False, help="Remove all the single line comments", type=bool)
+    parser.add_argument("--rm_block_comments", default=False, help="Remove all the block comments", type=bool)
+    args = parser.parse_args()
+
+    # initalize logger
+    logging.basicConfig(
+        format='%(asctime)s %(levelname)-8s %(message)s',
+        level=logging.INFO,
+        datefmt='%Y-%m-%d %H:%M:%S'
+    )
+    logger = logging.getLogger()
+    logger.setLevel(logging.DEBUG)
+
+    # create anonymizer class
+    anonymizer = Anonymizer(logger, args.rm_single_comments, args.rm_block_comments)
+
+    # add anonymizer options
+    if ("all" in args.add):
+        anonymizer.anonDicts = Regex_Dict
+        logger.info("Added all the anonymizer types")
+    else: 
+        for anon in args.add:
+            anonymizer.addDicts(anon)
+
+    # obtain files from test dir
+    test_dir = Path("Test").rglob("*")
```

Comment:
Add as possible flag with default value instead of hardcoded?

Reply:
I have modified this. So users can add the respective file path if they want.